src/polysolve/nonlinear/Criteria.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <cstddef>
+#include <iostream>
+
+namespace polysolve::nonlinear
+{
+    // Source: https://github.com/PatWie/CppNumericalSolvers/blob/7eddf28fa5a8872a956d3c8666055cac2f5a535d/include/cppoptlib/meta.h
+    // License: MIT
+
+    enum class Status
+    {
+        NotStarted = -1,
+        Continue = 0,
+        IterationLimit,
+        XDeltaTolerance,
+        FDeltaTolerance,
+        GradNormTolerance,
+        Condition,
+        UserDefined
+    };
+
+    template <typename T>
+    class Criteria
+    {
+    public:
+        size_t iterations; //!< Maximum number of iterations
+        T xDelta;          //!< Minimum change in parameter vector
+        T fDelta;          //!< Minimum change in cost function
+        T gradNorm;        //!< Minimum norm of gradient vector
+        T condition;       //!< Maximum condition number of Hessian
```

Comment:
we don't need conditions, and we can rename the others using the underscore we use everywhere.

also we can add the additional stuff we have like characteristic_length, use_grad_norm_tol, allow_out_of_iterations, etc.

Reply:
I removed the condition but left them without the underscore because they are public variables.